### PR TITLE
Test for positive case of upgrade process

### PR DESCRIPTION
**Pull request by anikitinDSR**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-826 Automate upgrade test in docker](https://app.clickup.com/t/6600954/DEV-826)

File: tests/upgrade/initiate_upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -euox pipefail
+
+. common.sh
+
+# Wait for start ordering, till height 2
+bash ../networks/wait_for_chain.sh 2
```
Comment (Toktar) [inline tests/upgrade/initiate_upgrade.sh:8]: Why do we need waiting?
Can we write `bank send` and `create-did` before upgrade?

File: tests/upgrade/upgrade_and_check.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# Wait for upgrade height
+bash ../networks/wait_for_chain.sh $(echo $UPGRADE_HEIGHT+2 | bc)
+
+CURRENT_VERSION=$(curl -s http://localhost:26657/abci_info | jq '.result.response.version' | grep -Eo '[0-9]*\.[0-9]*\.[0-9]*')
+
+if [ $CURRENT_VERSION != $CHEQD_VERSION_TO ] ; then
+     echo "Upgrade to version $CHEQD_VERSION_TO was not successful"
+     exit 1
+fi
+
```
[the remark this answers is not in the document]
Comment (anikitinDSR) [reply tests/upgrade/upgrade_and_check.sh:30]: Yes, we can.

File: tests/upgrade/initiate_upgrade.sh (hunk 1 above)
Comment (anikitinDSR) [reply tests/upgrade/initiate_upgrade.sh:8]: Yes, we can.

File: tests/upgrade/gen_node_configs.sh [hunk 3]
```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (anikitinDSR) [reply tests/upgrade/gen_node_configs.sh:1]: I had to play with another genesis parameters. If it's important we can try to combine them

**Review by ankurdotb — APPROVED** [review]

**Pull request merged**